### PR TITLE
SoundWire: add quirks for HP Omen 16-k0005TX

### PR DESCRIPTION
**Pull request by plbossart**
Set of patches to get audio support on HP Omen 16 SoundWire device

BugLink: https://github.com/thesofproject/linux/issues/3727

Companion topology fix:
https://github.com/thesofproject/sof/pull/5974

We still need to check the jack detection (JD1 v. JD2)

### CODE REVIEW

**Comment by greg-intel** [comment]
SOFCI TEST
There was a CI outage, so the tests were interrupted.  Note that they may take longer than usual and be missing a few devices because a few other tests are currently running.

File: drivers/soundwire/dmi-quirks.c [hunk 1]
```diff
@@ -55,7 +55,26 @@ static const struct adr_remap dell_sku_0A3E[] = {
 	{}
 };
 
+/*
+ * The HP Omen 16-k0005TX does not expose the correct version of RT711
+ * and does not expose a RT1316 on link1
```
Comment (ranj063) [inline drivers/soundwire/dmi-quirks.c]: did you mean link3 here?

Comment (plbossart) [reply drivers/soundwire/dmi-quirks.c]: yes, old comment. Need to update this.

Comment (plbossart) [reply drivers/soundwire/dmi-quirks.c]: fixed, thanks @ranj063 for catching this.

**Review by ranj063 — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Comment by plbossart** [comment]
SOFCI TEST

**Pull request merged**